File: Ad-Hoc/pagina15/2444/2444.py
volume_minimo, volume_maximo = 0, 100 #volume da TV

volume_inicial, quantidade_trocas = [int(x) for x in input().split()]
alteracoes_por_troca = [int(x) for x in input().split()]

volume_atual = volume_inicial
for i in range(quantidade_trocas):
    #Vnovo = max(0, min(100, Vatual + variacao))
    variacao = alteracoes_por_troca[i]
    novo_volume = max(0, min(100, volume_atual + variacao))
    volume_atual = novo_volume
    
print(volume_atual)


# VA = 50, delta = 14
# Vnovo = max(0, min(100, (50 + 14)))
# Vnovo = max(0, min(100, 64))
# Vnovo = max(0, 64)
# Vnovo = 64

File: Ad-Hoc/pagina15/2444/2444.ts

const fs = require("fs")
const input = fs.readFileSync("/dev/stdin", "utf8").split("\n")
const [volume_inicial, qnt_alteracoes] = input.shift().split(" ").map(Number)
const alteracoes = input.shift().split(" ").map(Number)
const vol_min = 0; const vol_max = 100;

let vol_atual = volume_inicial;
for(let i = 0; i < qnt_alteracoes; i++){
    let variacao = alteracoes[i]
    let novo_vol = Math.max(vol_min, Math.min(vol_max, vol_atual + variacao))
    vol_atual = novo_vol
}

console.log(vol_atual)